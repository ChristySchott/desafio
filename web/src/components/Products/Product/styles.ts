import styled from 'styled-components';

export const Container = styled.li`
  margin-top: 25px;
  width: 135px;

  @media (min-width: 768px) {
    width: 150px;
  }

  @media (min-width: 1024px) {
    width: 170px;
  }
`;

export const Image = styled.div`
  border: 1px solid #ddd;

  img {
    width: 100%;
  }
`;

export const Infos = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  align-items: center;
  text-align: center;

  height: 100px;
  padding: 10px 0;

  span {
    text-transform: uppercase;
    font-size: 1.6rem;
  }

  span:nth-child(2) {
    text-decoration: line-through;
    color: #999;
    font-size: 1.5rem;
  }

  strong {
    color: #c61a28;
    font-weight: 600;
  }
`;
