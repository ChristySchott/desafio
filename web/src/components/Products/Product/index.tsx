/* eslint-disable import/no-dynamic-require */
/* eslint-disable global-require */
import React from 'react';

import formatValue from 'utils/formatValue';
import { ProductInterface, useCart } from 'hooks/cart';
import { Container, Image, Infos } from './styles';

interface ProductProps {
  imageUrl: string;
  alt: string;
  name: string;
  offer?: number;
  price: number;
  item: ProductInterface;
}

const Product: React.FC<ProductProps> = ({ alt, name, offer, price, item }) => {
  const { addToCart } = useCart();

  function handleAddToCart(product: ProductInterface): void {
    addToCart(product);
  }

  return (
    <Container>
      <a href="/">
        <Image>
          <img src={require('assets/products/shoes-6.jpg')} alt={alt} />
        </Image>
      </a>

      <Infos>
        <span>{name}</span>
        {!!offer && <span>{formatValue(offer)}</span>}
        <strong>{formatValue(price)}</strong>
      </Infos>

      <button type="button" onClick={() => handleAddToCart(item)}>
        Comprar
      </button>
    </Container>
  );
};

export default Product;
