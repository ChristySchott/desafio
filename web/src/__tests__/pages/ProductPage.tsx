import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import { render } from '@testing-library/react';

import Product from '../../pages/Product';

describe('Product Page', () => {
  it('should render with no error', () => {
    const { container } = render(
      <BrowserRouter>
        <Product />
      </BrowserRouter>,
    );

    expect(container).toBeTruthy();

    expect(container.firstChild).toMatchSnapshot();
  });
});
