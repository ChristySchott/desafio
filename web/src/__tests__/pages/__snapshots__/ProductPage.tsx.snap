// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product Page should render with no error 1`] = `<div />`;
